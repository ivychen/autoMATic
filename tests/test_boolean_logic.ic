int main() {
	int x = 0;
	int y = 2;

	if (x == 0) print("x == 0");

	bool pred = (x == 0);
	if (pred) print("x == 0");
	if (not not pred) print("x == 0");

	pred = not pred;
	if (pred == false) print("pred is false");

	pred = true;
	if (pred == true) print("pred is true");

	bool pred2 = false;
	if (pred and pred2) print("pred and pred2 are true"); // false
	pred2 = true;
	if (pred and pred2) print("pred and pred2 are true"); // true

	if (not pred and not pred2) print("!pred and !pred2 are true"); // false
	if (not not pred and not not pred2) print("!!pred and !!pred2 are true"); // true

	return 0;
}
