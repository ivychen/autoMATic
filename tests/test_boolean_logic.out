x == 0
x == 0
x == 0
pred is false
